sunpy/util/util.py

```diff
@@ -212,3 +212,37 @@ def expand_list_generator(inp):
                 yield nested_item
         else:
             yield item
+
+def partial_match(key, dictionary):
+    """
+    Return value/values based on the partial key. Each element of the partial key is matched against the keys of the
+    dictionary and if a match is found the value of the key is returned. Even a partial match works here i.e 
+    even if the key matches partially a value is returned. Note that this function uses a generator so we need to 
+    type cast the output to a list to get the values. The example below will return a list containing the values `[1,2]` 
+    as the given key partially matches with two keys i.e ('abc','def','ghi') and ('abc','def','xyz') in the dictionary.
```

Comment:
You should line break between the first sentence and the rest of the text here. 

I think the last line can be removed.